### ISSUE TITLE
Nodogsplash stopped working when trying to authenticate with an expired/invalid token

### ISSUE THREAD

**Issue by ClydeAmazing** [nodogsplash/nodogsplash#520]
Hello,

When accessing http://{nds_gateway}:2050/nodogsplash_auth/?tok={expired/invalid token} makes NDS stop working, while produces "Token is invalid." error in the background before exiting.

This only occurs when client is not yet authenticated otherwise nds will successfully render the status page.

I am using NDS ver. 4.4.1 beta running on Raspberry PI 3

**Comment by bluewavenet** [comment]
@ClydeAmazing 
Are you running Raspbian or OpenWrt?

Note that you should never access the nds gateway directly. You should be redirected there by the client CPD.
If you do access it directly you will most likely get no response or an error.

Note also that your PI 3 MUST be configured as a wireless router and not as a wireless access point.

Can you please post details of your configuration.

**Comment by ClydeAmazing** [comment]
@bluewavenet 
I am running Raspbian, and RPi is configured as a router and is working quite nicely.
Another web application is also running within the same device on a different port.

`FirewallRuleSet preauthenticated-users {
  FirewallRule allow tcp port 8080 to 10.0.0.1
}`

I am aware that the nds should not be accessed directly, but I also usually access my web application manually through the browser by typing http://10.0.0.1:8080

However, browsers tends to autocomplete based on previous search entries and can also be previous nds authentication request (e.g. http://10.0.0.1:2050/nodogsplash_auth/?tok=f624d095) which is sometimes accidentally entered instead.

I guess it is expected to display an error if I accidentally entered http://10.0.0.1:2050/nodogsplash_auth/?tok={expired_token}, but it returns connection timeout and causes NDS to stop working in the background. (verified by running ndsctl status which returns `ndsctl: nodogsplash probably not started`)

I would like to stress out that this only occurs when the device requesting the expired authentication request is not yet authenticated by the NDS (Preauthenticated).

Attached is the rest of the details of my nds config.
[nodogsplash.txt](https://github.com/nodogsplash/nodogsplash/files/4177487/nodogsplash.txt)

**Comment by bluewavenet** [comment]
@ClydeAmazing 
I just tried it here on v4.4.1beta and cannot replicate the issue (albeit on OpenWrt)
I get "Unable to connect" in the browser as expected and afterwards NDS is still running.

However - you should never get anything like:
`http://10.0.0.1:2050/nodogsplash_auth/?tok=f624d095)`
in your browser history as it is the target of the form action on the splash page.
Maybe try clearing your history in case you have entered it manually at some time.

This does not explain why NDS is shutting down or crashing for you.
Is there anything in the system log?

Try running in the foreground to see if you get any indications.

**Comment by ClydeAmazing** [comment]
@bluewavenet 

I tried clearing the browser as you have mentioned but still the problem persists.

I did somehow managed to solved the problem by properly installing Nodogsplash by creating a .deb file.
Previously, i run NDS by creating ang entry in rc.local to automatically run on startup (which was from an old tutorial I found somewhere).

I am not experiencing the problem anymore since then.

Thank you so much for your assistance.
